.github/workflows/notify_ms_teams.yml

```diff
@@ -0,0 +1,72 @@
+name: Notify Teams on Push
+on:
+  push:
+    branches:
+      - main
+
+env:
+  environment: dev
+
+jobs:
+  notify-teams:
+    runs-on: ubuntu-latest
+    environment:
+      name: dev
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Extract Commit Information
+        id: commit_info
+        run: |
+          COMMIT_MESSAGE=$(git log --format=%B -n 1 ${{ github.sha }})
+          COMMIT_SHA=$(git log --format=%H --abbrev=7 -n 1 ${{ github.sha }})
+          COMMIT_TIMESTAMP=$(git log --format=%ci -n 1 ${{ github.sha }})
+          COMMIT_AUTHOR=${{ github.actor }}
+
+          echo "Commit Message: $COMMIT_MESSAGE"
+          echo "Commit SHA: $COMMIT_SHA"
+          echo "Commit Author: $COMMIT_AUTHOR"
+          echo "Commit Timestamp: $COMMIT_TIMESTAMP"
+
+          echo "COMMIT_MESSAGE=$COMMIT_MESSAGE" >> $GITHUB_ENV
```

Comment:
Not a blocker, would it help to reduce this to single step which then will not require any env storage?

Reply:
I wanted to split up the steps into readable sections here (even if it meant using env storage!)

Reply:
Makes sense, thanks.